### PR TITLE
PIM-4882: Cannot import products into a variant group if an axis is a reference data

### PR DESCRIPTION
**Pull request by ClementGautier**
| Q                  | A
| ------------------ | ---
| Migration script?  | N
| Behats             | Y
| Specs & Static     | Y
| Changelog updated  | Y

### CODE REVIEW

File: src/Pim/Bundle/CatalogBundle/Doctrine/ORM/Repository/ProductRepository.php [hunk 1]
```diff
@@ -171,17 +171,31 @@ public function findAllForVariantGroup(GroupInterface $variantGroup, array $crit
...
+            }
+
+            if (isset($item['referenceData'])) {
+                ++$index;
+                $qb->andWhere(sprintf('Value_%s.%s = ?%d', $code, $item['referenceData']['name'], $index))
+                    ->setParameter($index, $item['referenceData']['data']);
+            }
         }
 
+        $toto = $qb->getQuery()->getSQL();
```
Comment (jjanvier) [inline src/Pim/Bundle/CatalogBundle/Doctrine/ORM/Repository/ProductRepository.php]: nope

File: src/Pim/Bundle/CatalogBundle/Validator/Constraints/UniqueVariantAxisValidator.php [hunk 2]
```diff
@@ -95,26 +95,23 @@ protected function validateVariantGroup(GroupInterface $variantGroup, Constraint
...
-                        $constraint,
-                        $variantGroup->getLabel(),
-                        implode(', ', $values)
-                    );
-                }
+        $criteria = $this->prepareQueryCriterias($group, $product, $constraint);
+        $matches = $this->getMatchingProducts($group, $product, $criteria);
+
+        if (count($matches) !== 0) {
+            $values = array();
```
Comment (jjanvier) [inline src/Pim/Bundle/CatalogBundle/Validator/Constraints/UniqueVariantAxisValidator.php]: we use the short notation

File: src/Pim/Bundle/CatalogBundle/Validator/Constraints/UniqueVariantAxisValidator.php [hunk 3]
```diff
@@ -149,6 +141,21 @@ protected function prepareQueryCriterias(
...
                 );
             }
+
+            $current = ['attribute' => $attribute];
+
+            if (null !== $value->getOption()) {
+                $current['option'] = $value->getOption();
+            }
+
+            if ($attribute->isBackendTypeReferenceData()) {
```
Comment (jjanvier) [inline src/Pim/Bundle/CatalogBundle/Validator/Constraints/UniqueVariantAxisValidator.php]: elseif?

File: src/Pim/Bundle/CatalogBundle/Doctrine/ORM/Repository/ProductRepository.php (hunk 1 above)
Comment (sumbobyboys) [inline src/Pim/Bundle/CatalogBundle/Doctrine/ORM/Repository/ProductRepository.php]: sure ?

**Comment by jjanvier** [comment]
gtm once fixed

File: src/Pim/Bundle/CatalogBundle/Doctrine/ORM/Repository/ProductRepository.php (hunk 1 above)
Comment (ClementGautier) [inline src/Pim/Bundle/CatalogBundle/Doctrine/ORM/Repository/ProductRepository.php]: xD good catch

**Comment by willy-ahva** [comment]
GTM

**Pull request merged**